vms/proposervm/block.go

```diff
@@ -141,13 +142,36 @@ func (p *postForkCommonComponents) Verify(
 			)
 		}
 
-		delay, err := p.verifyBlockDelay(ctx, parentTimestamp, parentPChainHeight, child)
-		if err != nil {
-			return err
+		shouldHaveProposer := true // post Durango this is always the case
+		if p.vm.IsDurangoActivated(parentTimestamp) {
+			err := p.verifyPostDurangoBlockDelay(ctx, parentTimestamp, parentPChainHeight, child)
+			if err != nil {
+				return err
+			}
+
+			// Post durango, p.vm.Ctx.NodeID slot may be so far in time that we don't
+			// schedule it when resetting vm preferred block. So we make sure to reschedule timer
+			// every time we verify a block to avoid stalls where innerVM is ready to build a block
+			// but proposerVM has not scheduled its timer.
+			nextStartTime, err := p.vm.resetPostDurangoScheduler(ctx, child, childPChainHeight)
+			switch err {
+			case nil:
+				p.vm.Scheduler.SetBuildBlockTime(nextStartTime)
+			case proposer.ErrNoSlotsScheduledInNextFuture:
+				// Can't schedule a slot now. We'll try later again
```

Comment:
Shouldn't we be scheduling to try to build after `MaxLookAheadWindow`? (Assuming this piece of code was supposed to be in `BuildBlock`.)